lib/app/dispatcher.rb

```diff
@@ -93,9 +105,36 @@ def start_polling_jobs!
         Utility::ExceptionTracking.log_exception(e, 'connector service failed due to unexpected error.')
       end
 
-      def send_heartbeat(connector_id, service_type)
-        Utility::Logger.info("Sending a heartbeat on [#{connector_id} - #{service_type}]...")
-        Core::Heartbeat.send(connector_id, service_type)
+      def start_heartbeat_task!
+        Thread.new do
+          Utility::Logger.info('Start heartbeat task...')
+          heartbeat_scheduler.when_triggered do |connector_settings|
+            pool.post do
+              service_type = connector_settings.service_type
+              connector_id = connector_settings.id
+              Utility::Logger.info("Sending a heartbeat for connector (ID: #{connector_id}, service type: #{service_type})...")
+              Core::Heartbeat.send(connector_id, service_type)
+            rescue StandardError => e
+              Utility::ExceptionTracking.log_exception(e, "Heartbeat task for connector (ID: #{connector_id}, service type: #{service_type}) failed due to unexpected error.")
+            end
+          rescue StandardError => e
+            Utility::ExceptionTracking.log_exception(e, 'connector service failed due to unexpected error.')
+          end
+        end
+      end
+
+      def start_configuration_task!
+        Thread.new do
+          Utility::Logger.info('Start configuration task...')
+          configuration_scheduler.when_triggered do |connector_settings|
```

Comment:
As far as I can see, the configuration scheduler [can only be triggered when the connector is in status `CREATED`](https://github.com/elastic/connectors-ruby/blob/692375321e20bda9596fa45a5a46e64b22334318/lib/core/scheduler.rb#L129). How many times can this happen in the life of the connector, and is it enough to start a thread per connector that would poll for it every 5 seconds?

Reply:
>How many times can this happen in the life of the connector
Once, after it's created.

I was trying hard to [get rid of this task in connector service](https://elastic.slack.com/archives/C03GAPTQ2JF/p1661505372210029?thread_ts=1661351497.576319&cid=C03GAPTQ2JF), but this is the solution at least for 8.5